sqlmodel/_compat.py

```diff
@@ -459,7 +468,9 @@ def is_field_noneable(field: "FieldInfo") -> bool:
         return field.allow_none  # type: ignore[no-any-return, attr-defined]
 
     def get_sa_type_from_field(field: Any) -> Any:
-        if isinstance(field.type_, type) and field.shape == SHAPE_SINGLETON:
+        if get_origin(field.type_) is Literal:
+            return AutoString
+        elif isinstance(field.type_, type) and field.shape == SHAPE_SINGLETON:
```

Comment:
```suggestion
            return Literal
        if isinstance(field.type_, type) and field.shape == SHAPE_SINGLETON:
```

How about making it similar to how it works with Pydantic V2?
This way we will not need `if isinstance(type_, type) and issubclass(type_, sa_types.TypeEngine):` in `get_sqlalchemy_type`